backend/lucene/src/main/java/org/hibernate/search/backend/lucene/types/codec/impl/AbstractLuceneNumericFieldCodec.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Hibernate Search, full-text search for your domain model
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.search.backend.lucene.types.codec.impl;
+
+import org.hibernate.search.backend.lucene.document.impl.LuceneDocumentBuilder;
+import org.hibernate.search.backend.lucene.util.impl.LuceneFields;
+
+import org.apache.lucene.index.Term;
+import org.apache.lucene.search.DocValuesFieldExistsQuery;
+import org.apache.lucene.search.Query;
+import org.apache.lucene.search.TermQuery;
+
+public abstract class AbstractLuceneNumericFieldCodec<F, E extends Number> implements LuceneStandardFieldCodec<F, E> {
+
+	private final boolean projectable;
+
+	private final boolean sortable;
+
+	public AbstractLuceneNumericFieldCodec(boolean projectable, boolean sortable) {
+		this.projectable = projectable;
+		this.sortable = sortable;
+	}
+
+	@Override
+	public final void encode(LuceneDocumentBuilder documentBuilder, String absoluteFieldPath, F value) {
+		if ( value == null ) {
+			return;
+		}
+
+		E encodedValue = encode( value );
+
+		if ( projectable ) {
+			doEncodeForProjection( documentBuilder, absoluteFieldPath, value, encodedValue );
+		}
+
+		LuceneNumericDomain<E> domain = getDomain();
+
+		if ( sortable ) {
+			documentBuilder.addField( domain.createDocValuesField( absoluteFieldPath, encodedValue ) );
+		}
+		else {
+			// For createExistsQuery()
```

Comment:
I agree with this optimization idea. Is there a way to apply it even if the field isn't sortable?

Reply:
No, we need docvalues in order to avoid indexing the field name. And currently we only have docvalues when the field is sortable.